extensions/arc/src/ui/components/keyValueContainer.ts

```diff
@@ -56,31 +54,52 @@ export abstract class KeyValue {
 		return container;
 	}
 
+	/** Refreshes the value of the KeyValue pair */
+	public abstract refresh(value: string): void;
+
 	/** Returns a component representing the value of the KeyValue pair */
 	protected abstract getValueComponent(modelBuilder: azdata.ModelBuilder): azdata.Component;
 }
 
 /** Implementation of KeyValue where the value is text */
 export class TextKeyValue extends KeyValue {
-	getValueComponent(modelBuilder: azdata.ModelBuilder): azdata.Component {
-		return modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+	private text?: azdata.TextComponent;
+
+	public refresh(value: string) {
+		if (this.text) {
+			this.text.value = value;
+		}
+	}
+
+	protected getValueComponent(modelBuilder: azdata.ModelBuilder): azdata.Component {
```

Comment:
I'd just create the component in the constructor and have it take in the modelBuilder then. That way you don't have to worry about the component not being created and multiple calls to getValueComponent creating multiple dangling components.

Reply:
Now using modelBuilder in the constructors.  Also disposing the onDidClick listeners.